### PR TITLE
✨ Do you use the EditorRequired attribute for required parameters in Blazor?

### PR DESCRIPTION
**Pull request by MattParkerDev**
@adamcogan @Hona @bradystroud 

New Rule for Blazor: `Do you use the EditorRequired attribute for required parameters in Blazor?`

### CODE REVIEW

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+<PageTitle>Home</PageTitle>
+
+<TestComponent />
+```
+
+::: good  
+Figure: Good example - My IDE will give me a warning that I have forgotten the Name parameter
+
+:::
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:71]: I think a screenshot of the warning would be better

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+archivedreason:
+title: Do you use the EditorRequired attribute for required parameters in Blazor?
+guid: 125bf5aa-7b54-4aa3-8f71-8178af185afc
+uri: editor-required-blazor-component-parameters
+created: 2023-08-26T00:00:00.0000000Z
+authors:
+  - title: Matt Parker
+    url: https://ssw.com.au/people/matt-parker
+related: []
+redirects:
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:12]:
```suggestion
redirects: []
```

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+type: rule
+archivedreason:
+title: Do you use the EditorRequired attribute for required parameters in Blazor?
+guid: 125bf5aa-7b54-4aa3-8f71-8178af185afc
+uri: editor-required-blazor-component-parameters
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:6]:
```suggestion
uri: editor-required-blazor
```

If you agree, change foldername as well

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+authors:
+  - title: Matt Parker
+    url: https://ssw.com.au/people/matt-parker
+related: []
+redirects:
+---
+
+When you create a Blazor component, view parameters are marked with the `[Parameter]` attribute to indicate that they must be supplied by the parent component. By default, this is not enforced, which may lead to errors where you forget to pass in parameters where you use the component.
+
+You should use the `[EditorRequired]` attribute to mark parameters that are required in your Blazor Component.
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:17]:
```suggestion
You should use the `[EditorRequired]` attribute to mark parameters that are required in your Blazor component.
```

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+::: good  
+Figure: Good example - My IDE will give me a warning that I have forgotten the Name parameter
+
+:::
+
+`Component 'TestComponent' expects a value for the parameter 'Name', but a value may not have been provided.`
+
+You should configure this warning (RZ2012) as an error so your IDE will fail to build if you are missing a required parameter. Add `<WarningsAsErrors>RZ2012</WarningsAsErrors>` to your Blazor .csproj file:
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:75]: Thoughts on WarningsAsErrors or all warnings?
I wonder if we already have a rule on that?

**Review by bradystroud — CHANGES_REQUESTED** [review]
Cool rule - i didnt know about this

File: rules/editor-required-blazor-component-parameters/rule.md (hunk 5 above)
Comment (MattParkerDev) [reply rules/editor-required-blazor-component-parameters/rule.md:75]: I'll link to the original rule

File: rules/editor-required-blazor-component-parameters/rule.md (hunk 3 above)
Comment (MattParkerDev) [reply rules/editor-required-blazor-component-parameters/rule.md:6]: As per convo with @bradystroud, happy to leave as is.

File: rules/editor-required-blazor-component-parameters/rule.md (hunk 1 above)
Comment (MattParkerDev) [reply rules/editor-required-blazor-component-parameters/rule.md]: Done, please review and resolve if happy.

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+```razor
+@page "/"
+
+<PageTitle>Home</PageTitle>
+
+<TestComponent />
+```
+
+::: bad  
+Figure: Bad example - I could forget to pass a variable to the Name property
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:43]:
```suggestion
Figure: Bad example - Developers could forget to pass a variable to the Name property
```

File: rules/editor-required-blazor-component-parameters/rule.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+```razor
+@page "/"
+
+<PageTitle>Home</PageTitle>
+
+<TestComponent />
+```
+
+::: good  
+![Figure: Good example - My IDE warns me that I have forgotten the Name parameter](ide-warning.png)
```
Comment (bradystroud) [inline rules/editor-required-blazor-component-parameters/rule.md:69]:
```suggestion
![Figure: Good example - The IDE warns developers if they forget the Name parameter](ide-warning.png)
```

**Review by bradystroud — APPROVED** [review]

**Pull request merged**